src/Indicator.vala

```diff
@@ -47,18 +48,53 @@ public class DateTime.Indicator : Wingpanel.Indicator {
     public override Gtk.Widget? get_widget () {
         if (main_grid == null) {
             calendar = new Widgets.Calendar ();
-            calendar.margin_top = 6;
-            calendar.margin_bottom = 6;
+
+            var provider = new Gtk.CssProvider ();
+            provider.load_from_resource ("/io/elementary/desktop/wingpanel/datetime/main.css");
 
             var settings_button = new Gtk.ModelButton ();
             settings_button.text = _("Date & Time Settings…");
 
+            var header_label = new Gtk.Label (_("Events"));
+            header_label.get_style_context ().add_class ("header-label");
+            header_label.get_style_context ().add_provider (provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+            header_label.halign = Gtk.Align.START;
+            header_label.width_chars = 13;
+            header_label.xalign = 0;
+
+            var cal_icon = new Gtk.Image.from_icon_name ("office-calendar", Gtk.IconSize.LARGE_TOOLBAR);
+            cal_icon.halign = Gtk.Align.END;
+            var cal_button = new Gtk.Button ();
+            cal_button.set_image (cal_icon);
+            cal_button.get_style_context ().add_class (Gtk.STYLE_CLASS_FLAT);
+            cal_button.get_style_context ().remove_class (Gtk.STYLE_CLASS_BUTTON);
+            cal_button.get_style_context ().remove_class ("text-button");
+            cal_button.set_tooltip_text (_("Open Calendar"));
+
+            var header_grid = new Gtk.Grid ();
+            header_grid.column_spacing = 6;
+            header_grid.valign = Gtk.Align.CENTER;
+            header_grid.margin = 6;
+            header_grid.attach (header_label, 0, 0);
+            header_grid.attach (cal_button, 1, 0);
+
+            var sep = new Gtk.Separator (Gtk.Orientation.VERTICAL);
+
+            //TRANSLATORS: Don't remove the "\n\t" part.
+            no_events_label = new Gtk.Label (_("No Events Scheduled\n\tIn This Day"));
+            no_events_label.get_style_context ().add_class (Granite.STYLE_CLASS_H3_LABEL);
+            no_events_label.expand = true;
+            no_events_label.sensitive = false;
```

Comment:
Please use `Gtk.STYLE_CLASS_DIM_LABEL` instead of setting sensitivity. These are tested for contrast differently